ui/component/app/index.js

```diff
@@ -10,56 +11,117 @@ import {
+import { selectModal, selectActiveChannelClaim, selectIsReloadRequired } from 'redux/selectors/app';
 import { selectUploadCount } from 'redux/selectors/publish';
 import { doSetLanguage } from 'redux/actions/settings';
 import { doSyncLoop } from 'redux/actions/sync';
-import { doDownloadUpgradeRequested, doSignIn, doSetIncognito } from 'redux/actions/app';
+import { doSignIn, doSetIncognito } from 'redux/actions/app';
 import { doFetchModBlockedList, doFetchCommentModAmIList } from 'redux/actions/comments';
+import { selectActiveLiveClaimForChannel } from 'redux/selectors/livestream';
+import { doFetchChannelLiveStatus } from 'redux/actions/livestream';
+import { normalizeURI } from 'util/lbryURI';
+import { generateGoogleCacheUrl } from 'util/url';
 import App from './view';
 
-const select = (state) => ({
-  user: selectUser(state),
-  locale: selectUserLocale(state),
-  theme: selectThemePath(state),
-  language: selectLanguage(state),
-  languages: selectLoadedLanguages(state),
-  autoUpdateDownloaded: selectAutoUpdateDownloaded(state),
-  isUpgradeAvailable: selectIsUpgradeAvailable(state),
-  isReloadRequired: selectIsReloadRequired(state),
-  syncError: selectGetSyncErrorMessage(state),
-  syncIsLocked: selectSyncIsLocked(state),
-  uploadCount: selectUploadCount(state),
-  rewards: selectUnclaimedRewards(state),
-  isAuthenticated: selectUserVerifiedEmail(state),
-  currentModal: selectModal(state),
-  syncFatalError: selectSyncFatalError(state),
-  activeChannelClaim: selectActiveChannelClaim(state),
-  myChannelClaimIds: selectMyChannelClaimIds(state),
-  hasPremiumPlus: selectOdyseeMembershipIsPremiumPlus(state),
-  homepageFetched: selectHomepageFetched(state),
-});
+const LATEST_PATH = `/$/${PAGES.LATEST}/`;
+const LIVE_PATH = `/$/${PAGES.LIVE_NOW}/`;
+const EMBED_PATH = `/$/${PAGES.EMBED}/`;
+
+const select = (state, props) => {
+  const { pathname, hash, search } = state.router.location;
+
+  const urlPath = pathname + hash;
+  const embedPath = urlPath.startsWith(EMBED_PATH);
+  const urlParams = new URLSearchParams(search);
+  const featureParam = urlParams.get('feature');
+  const latestContentPath = urlPath.startsWith(LATEST_PATH) || (embedPath && featureParam === PAGES.LATEST);
+  const liveContentPath = urlPath.startsWith(LIVE_PATH) || (embedPath && featureParam === PAGES.LIVE_NOW);
+  const isNewestPath = latestContentPath || liveContentPath;
+
+  let path;
+  if (isNewestPath) {
+    path = urlPath.replace(embedPath ? EMBED_PATH : latestContentPath ? LATEST_PATH : LIVE_PATH, '');
+  } else {
+    // Remove the leading "/" added by the browser
+    path = urlPath.slice(1);
+  }
+  path = path.replace(/:/g, '#');
+
+  if (isNewestPath && !path.startsWith('@')) {
+    path = `@${path}`;
+  }
+
+  if (search && search.startsWith('?q=cache:')) {
+    generateGoogleCacheUrl(search, path);
+  }
+
+  let uri;
+  try {
+    uri = normalizeURI(path);
```

Comment:
Just something to take note of when testing (performance).

The `index.js` gets hit a lot (while the jsx only gets hit if the prop changes), especially for this component, and `normalizeURI` is one of the slow functions due to `parseURI`.